exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/join/TestMergeJoinAdvanced.java

```diff
@@ -255,6 +260,42 @@ public void testDrill4196() throws Exception {
       .go();
   }
 
+  @Test // DRILL-6491
+  public void testMergeIsNotSelectedForFullJoin() throws Exception {
+    setSessionOption(PlannerSettings.HASHJOIN.getOptionName(), true);
+
+    String query = "select * " +
+        " from (select employee_id from cp.`employee.json` order by employee_id) e1 " +
+        " full outer join (select employee_id from cp.`employee.json` order by employee_id) e2 " +
+        " on e1.employee_id = e2.employee_id " +
+        " limit 10";
+    try {
+      testPlanMatchingPatterns(query, new String[] {"HashJoin"}, new String[] {"MergeJoin"});
+    } finally {
+      test("ALTER SESSION RESET ALL");
+    }
+  }
+
+  @Test // DRILL-6491
+  public void testFullJoinIsNotSupported() throws Exception {
+    thrown.expect(UserRemoteException.class);
+    thrown.expectMessage(CoreMatchers.containsString("SYSTEM ERROR: CannotPlanException"));
+
+    setSessionOption(PlannerSettings.HASHJOIN.getOptionName(), false);
+    setSessionOption(PlannerSettings.NESTEDLOOPJOIN.getOptionName(), false);
+
+    String query = "select * " +
+        " from (select employee_id from cp.`employee.json` order by employee_id) e1 " +
+        " full outer join (select employee_id from cp.`employee.json` order by employee_id) e2 " +
+        " on e1.employee_id = e2.employee_id " +
+        " limit 10";
+    try {
+      test(query);
+    } finally {
+      test("ALTER SESSION RESET ALL");
```

Comment:
Do not reset all sessions options.

Reply:
Do we need finally if we have thrown?

Reply:
Yes, we still need it as the Exception is thrown for real (though, expected).